projects/hr-portal/src/app/hr-portal/hr-portal.tsx

```diff
@@ -0,0 +1,202 @@
+import "./hr-portal.scss";
+
+import {
+  IgrTreeGrid,
+  IgrGridToolbar,
+  IgrGridToolbarTitle,
+  IgrGridToolbarActions,
+  IgrGridToolbarAdvancedFiltering,
+  IgrGridToolbarHiding,
+  IgrGridToolbarPinning,
+  IgrGridToolbarExporter,
+  IgrColumn,
+  IgrPaginator,
+  IgrCellTemplateContext,
+} from "igniteui-react-grids";
+import {
+  IgrIcon,
+  IgrAvatar,
+  IgrIconButton,
+  IgrButton,
+  registerIcon,
+} from "igniteui-react";
+import "igniteui-react-grids/grids/combined";
+import "igniteui-react-grids/grids/themes/light/fluent.css";
+import { dataService } from "./../services/data.service";
+import { icons } from "../data/icons/Icons";
+
+export default function HRPortal() {
+  const [data, setData] = useState<any[]>([]);
+  const [isSorted, setIsSorted] = useState(false);
+  const gridRef = useRef<IgrTreeGrid | null>(null);
+
+  useEffect(() => {
+    dataService.fetchData().then((fetchedData: any) => {
+      setData(fetchedData);
+    });
+
+    icons.forEach(
+      (element: { name: string; path: string; category: string }) => {
+        const normalizedPath = element.path.startsWith("/")
+          ? `${import.meta.env.BASE_URL}${element.path.slice(1)}`
+          : `${import.meta.env.BASE_URL}${element.path}`;
+
+        registerIcon(element.name, normalizedPath, element.category);
+      }
+    );
+  }, []);
+
+  const avatarTemplate = ({ cell }: IgrCellTemplateContext) => {
+    const data = cell.row.data;
+    return (
+      <div className="employeeDiv">
+        <IgrAvatar shape="rounded" src={data.Picture} className="small" />
+        <span>{data.Name}</span>
+      </div>
+    );
+  };
+
+  const countryIconTemplate = ({ cell }: IgrCellTemplateContext) => {
+    const data = cell.row?.data;
+    if (!data) {
+      return <div>No Data</div>; // Fallback for missing data
+    }
+    return (
+      <div className="flagDiv">
+        <IgrIcon collection="country-icons" name={data.Country} />
+        <span>
+          {data.Location}, {data.Country}
+        </span>
+      </div>
+    );
+  };
+
+  const contactsTemplate = ({ cell }: IgrCellTemplateContext) => {
+    const data = cell.row.data;
+    return (
+      <div className="center-content small">
+        <a href={`mailto:${data.Email}`}>
+          <IgrIconButton collection="hr-icons" name="mail" variant="flat" />
+        </a>
+        <a href={`tel:${data.Phone}`}>
+          <IgrIconButton collection="hr-icons" name="tel" variant="flat" />
+        </a>
+        <a href="https://www.linkedin.com" target="_blank" rel="noreferrer">
+          <IgrIconButton collection="hr-icons" name="linkedIn" variant="flat" />
+        </a>
+      </div>
+    );
+  };
+
+  const dateTemplate = ({ cell }: IgrCellTemplateContext) => {
+    const data = cell.row.data;
+    const formattedDate = new Date(data.HireDate).toISOString().split("T")[0];
+    return <>{formattedDate}</>;
+  };
+
+  const clearSorting = useCallback(() => {
```

Comment:
What about this `useCallback` again?

Reply:
removed it again, slipped when i was restoring the previous state of the file